### ISSUE TITLE
Kobo Sync token contents do not follow the expected json schema

### ISSUE THREAD

**Issue by wietsedv** [janeczku/calibre-web#3017]
**Describe the bug/problem** 

Since a couple of weeks synchronisation with Kobo does not work anymore and if I keep wifi on on my Kobo, the battery drains in a day. It appears that the Kobo Sync Token that Kobo gives in request headers cannot be parsed by Calibre Web. Calibre-Web nor Kobo have been updated since it last worked, so I am assuming that this will be hard to reproduce.

Has anyone experienced a similar issue before?

I have tried:

- Bypass reverse proxy with ip:port.
- Reset kobo sync database (app.db)
- Reset kobo sync token
- Kobo factory reset

**Logfile**

Endless repetitions of the following sequence:

```
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,085] DEBUG {cps.ub:84} Found stored session: 3a6e8f8f10293033f41dccff03ac2440f01b000811d699d88974030560913c5941a3473bc284e086c4c0240b81a3c2ab4185443ae363b4618aabe532c0fba268
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,085] ERROR {cps.services.SyncToken:126} Sync token contents do not follow the expected json schema.
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,085]  INFO {cps.kobo:143} Kobo library sync request received.
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,085] DEBUG {cps.kobo:144} SyncToken: 0001-01-01 00:00:00,0001-01-01 00:00:00,0001-01-01 00:00:00,0001-01-01 00:00:00,0001-01-01 00:00:00,
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,085] DEBUG {cps.kobo:145} Download link format http://192.168.0.1:8083/kobo/6d559739f510d466c496de21e2aa87a3/download/[bookid]/[bookformat]
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,085] DEBUG {cps.kobo:147} Kobo: Received unproxied request, changed request port to external server port
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,097] DEBUG {cps.kobo:204} Books to Sync: 0
Mar 10 11:32:17 homelab calibre-web[4193091]: [2024-03-10 11:32:17,101] DEBUG {cps.kobo:258} Remaining books to Sync: 0
```

**Environment (please complete the following information):**

 - OS: Linux (NixOS
 - Python version: 3.11.8
 - Calibre-Web version: 0.6.21 (from nixos-unstable)
 - Docker container:  n/a
 - Special Hardware: 64-bit
 - Browser: n/a/

**Comment by wietsedv** [comment]
Of course I managed to identify the problem just minutes after posting this issue. I do not have a definitive solution, but my issue is caused by the calibre-web distribution of `nixos-unstable`. When I switch to the `nixos-2311` package, the problem is solved. I will look into fixing the `nixos-unstable` version later, but since both are calibre-web 0.6.21, it is not a real calibre-web issue. Might be something about python or other package version compatibility.

**Comment by mbeutelspacher** [comment]
I experience the same error both under NixOS 24.05 as well as unstable. @wietsedv  were you able to resolve the problem?

**Comment by wietsedv** [comment]
I have solved the problem but I cannot remember how. When I opened this issue, it was broken on 23.11 but worked with the unstable package. After 24.05 was released, it was broken on both 24.05 and unstable I think. Currently, I'm using the 24.05 nixos module with the unstable package. There is nothing special in my nixos config. I use calibre-web through a traefik reverse proxy, but the problem was not affected by this.

My guess is that I changed something in the calibre-web settings page. I have vague memories of having to explicitly set an external port (either "Server Configuration" > "Server Port" -> 8083, or "Feature Configuration" > "Server External Port (for port forwarded API calls)" -> 8083)

Let me know if this does anything. This issue has taken me some time to solve and I'm sorry that I don't remember the cause or how I figured out the solution.

**Comment by mbeutelspacher** [comment]
I digged into that in the past and stumbed on weird issues with regards to byte-strings vs strings but never ended up fixing it.
However in the current 24.11 version everything ended up working after ensuring  to follow
https://github.com/janeczku/calibre-web/wiki/Kobo-Integration#additional-hints
